### PR TITLE
VDS-334: Create/update customer segment rule

### PR DESCRIPTION
**Pull request by trueboroda**
Without indexation.

### CODE REVIEW

File: src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-detail.js [hunk 1]
```diff
@@ -1,17 +1,46 @@
...
         const blade = $scope.blade;
         blade.headIcon = 'fa-pie-chart';
-        blade.isLoading = false;
         blade.activeBladeId = null;
-
         blade.currentEntity = {};
 
         $scope.groups = settings.getValues({ id: 'Customer.MemberGroups' });
 
+        blade.refresh = function (parentRefresh) {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-detail.js:11]: Make all functions arrowed if it is possible for unifying

File: src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-detail.tpl.html [hunk 2]
```diff
@@ -21,7 +21,7 @@
                     <div class="form-group __info list">
                         <label class="form-label">{{ 'demoCustomerSegmentsModule.blades.customer-segment-detail.user-group' | translate }}  <a href="" ng-click="openGroupsDictionarySettingManagement()" class="form-edit" va-permission="platform:setting:update"><i class="form-ico fa fa-pencil"></i></a></label>
                         <div class="form-input" data-role="input-control">
-                            <ui-select ng-model="blade.currentEntity.group">
+                            <ui-select ng-model="blade.currentEntity.userGroup">
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-detail.tpl.html:24]: Add required attribute

File: src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-list.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+                if ($scope.pageSettings.currentPage !== 1) {
+                    $scope.pageSettings.currentPage = 1;
+                }
+
+                customerSegmentsApi.search(getSearchCriteria(), function (data) {
+                    blade.isLoading = false;
+                    $scope.pageSettings.totalItems = data.totalCount;
+                    blade.currentEntities = data.results;
+                    $scope.hasMore = data.results.length === $scope.pageSettings.itemsPerPageCount;
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-list.js:19]: data.results && data.results.length

File: src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-list.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
+angular.module('virtoCommerce.DemoCustomerSegmentsModule')
+    .controller('virtoCommerce.DemoCustomerSegmentsModule.customerSegmentListController', ['$scope', 'platformWebApp.dialogService', 'platformWebApp.bladeUtils', 'platformWebApp.uiGridHelper', 'virtoCommerce.DemoCustomerSegmentsModule.customerSegmentsApi',
+        function ($scope, dialogService, bladeUtils, uiGridHelper, customerSegmentsApi) {
+            const blade = $scope.blade;
+            blade.headIcon = 'fa-pie-chart';
+            const bladeNavigationService = bladeUtils.bladeNavigationService;
+
+            blade.refresh = function () {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-list.js:8]: Use arrow function as much as you can.

File: src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/resources/customerSegments.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ angular.module('virtoCommerce.DemoCustomerSegmentsModule')
 .factory('virtoCommerce.DemoCustomerSegmentsModule.customerSegmentsApi', ['$resource', function ($resource) {
     return $resource('api/demo/customersegments/:id', {}, {
         new: { method: 'GET', url: 'api/demo/customersegments/new' },
-        update: { method: 'PUT' },
+        update: { method: 'POST', isArray: true },
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/resources/customerSegments.js:5]: save

**Review by pushnitsa — CHANGES_REQUESTED** [review]

**Review by pushnitsa — APPROVED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**